File: 2 semestre/Laboratório de Desenvolvimento de Algoritmos/Lista de Exercícios 4/src/Ex7.java
//Importando Scanner
import java.util.Scanner;

public class Ex7 {

	public static void main(String[] args){
		
		//Instanciando Scanner
		Scanner ent = new Scanner(System.in);
		
		System.out.println("Vetores Intercalados\n\n");

		//vars
		int a[] = new int[20];
		int b[] = new int[20];
		int res[] = new int[40];
		int i, iA=0, iB=0;
		
		//Entrada
		for( i=0 ; i<20; i++)
		{
			System.out.print("Digite a[" + i + "]: ");
			a[i] = ent.nextInt();
		}
		
		for( i=0 ; i<20; i++)
		{
			System.out.print("Digite b[" + i + "]: ");
			b[i] = ent.nextInt();
		}
		
		//Processamento
		for (i=0; i<40; i++)
		{
			if( i%2 == 0)
			{
				res[i] = a[iA];
				iA++;
			}
			else
			{
				res[i] = b[iB];
				iB++;
			}
		}
		
		//Saída
		System.out.println("\nVetor resultante:");
		for( i=0; i<40; i++)
		{
			System.out.printf("%d  ", res[i]);
		}
		
	}
}

File: 2 semestre/Técnica de Desenvolvimento de Algoritmos/Listas de Exercicios/Lista 4/ex7.c
#include <stdio.h>

main()
{
    printf("Vetores intercalados\n\n");

    //vars
    int v1[20], v2[20], res[40], i, contV1=0, contV2=0;

    //Entrada
    for( i=0; i<20; i++)
    {
        printf("Digite v1[%d]: ", i);
        scanf("%d", &v1[i]);
    }

    for( i=0; i<20; i++ )
    {
        printf("Digite v2[%d]: ", i);
        scanf("%d", &v2[i]);
    }

    //Processamento
    for( i=0; i<40; i++)
    {
        if( i%2 == 0)
        {
            res[i] = v1[contV1];
            contV1++;
        }
        else
        {
            res[i] = v2[contV2];
            contV2++;
        }
    }

    //Saída
    printf("\nVetor Resultante: \n\n");
    for( i=0; i<40; i++)
    {
        printf("%d\n", res[i]);
    }

}
